internal/ethapi/tracer.go

```diff
@@ -213,20 +215,26 @@ type JavascriptTracer struct {
 	contract      *contractWrapper       // Wrapper around the contract object
 	contractvalue otto.Value             // JS view of `contract`
 	err           error                  // Error, if one has occurred
+	result        interface{}            // Final result to return to the user
 }
 
 // NewJavascriptTracer instantiates a new JavascriptTracer instance.
 // code specifies a Javascript snippet, which must evaluate to an expression
 // returning an object with 'step' and 'result' functions.
 func NewJavascriptTracer(code string) (*JavascriptTracer, error) {
-	vm := otto.New()
-	vm.Interrupt = make(chan func(), 1)
+	jsvm := otto.New()
+	jsvm.Interrupt = make(chan func(), 1)
 
 	// Set up builtins for this environment
-	vm.Set("big", &fakeBig{})
-	vm.Set("toHex", hexutil.Encode)
-
-	jstracer, err := vm.Object("(" + code + ")")
+	jsvm.Set("big", &fakeBig{})
+	jsvm.Set("toHex", hexutil.Encode)
+	jsvm.Set("toAddress", common.BytesToAddress)
+	jsvm.Set("isPrecompiled", func(addr []byte) bool {
+		_, ok := vm.PrecompiledContractsByzantium[common.BytesToAddress(addr)]
```

Comment:
You can't assume all traces will be on byzantium!

Reply:
But the precompiles that don't exist won't be called on non-byzantium chains, and if they are, won't do anything interesting.